vmdb/app/models/mixins/retirement_mixin.rb

```diff
@@ -188,4 +187,18 @@ def retiring?
   def error_retiring?
     retirement_state == ERROR_RETIRING
   end
+
+  private
+
+  def setup_event_hash(requester)
+    event_hash = {:retirement_initiator => "system"}
+    event_hash[retirement_base_model_name.underscore.to_sym] = self
+    event_hash[:host] = host if self.respond_to?(:host)
+    if requester
+      event_hash[:user_id] = requester
+      event_hash[:retirement_initiator] = "user"
+    end
+    event_hash[:type] ||= self.class.name
+    event_hash
```

Comment:
@Fryguy 
It could end up in the queue but only as string parameters, except for host which looks like an AR object. The string parameters become automate attributes and during retries it will get re-queued.
e.g.

/System/Process/Event?MiqServer%3A%3Amiq_server=3&VmOrTemplate%3A%3Avm=1527&event_type=request_vm_retire      &object_name=Event&retirement_initiator=user&type=VmAmazon&user_id=admin&vm_id=1527&vmdb_objec     t_type=vm